### PR TITLE
[FluentIcon] Minor cleanup and clarification

### PR DESCRIPTION
**Pull request by c0g1t8**
# Pull Request

## 📖 Description

- Minor cleanup on code rendered for `FluentIcon`. Removes invalid `width` attribute from `SVG` style when `Width` set to `string.Empty`
- Update documentation to clarify behavior when `Width` set to `string.Empty`.
 
### 🎫 Issues

For responsive layout, it is preferable to use CSS versus embedding the style in HTML. The `width` of the SVG is embedded unless `FluentIcon`'s `Width` is explicity set to `string.Empty`

## 👩‍💻 Reviewer Notes

Any suggestions to the update on the documentation page? I wanted to add some information about this behavior without getting into a specific use case.

### General

- [ ] I have added tests for my changes.
- [X] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [X] I have read the [CONTRIBUTING](https://github.com/Microsoft/fluentui-blazor/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have added [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for my new compontent
- [X] I have modified an existing component
- [x] I have validated the [Unit Tests](https://github.com/Microsoft/fluentui-blazor/blob/master/unit-tests.md) for an existing component

### CODE REVIEW

File: examples/Demo/Shared/Pages/Icon/IconPage.razor [hunk 1]
```diff
@@ -94,7 +94,9 @@
             <code>&lt;FluentButton IconStart="@@(new MyCircle())">Refresh&lt;/FluentButton></code>
         </li>
     </ul>
-    <br /><br />
+    Setting <code>Width</code> to <code>string.Empty</code> makes the icon 100% width of the enclosing container and stylable from CSS.
```
Comment (dvoituron) [inline examples/Demo/Shared/Pages/Icon/IconPage.razor:97]: Probably better to explain:

> Setting <code>Width=""</code> (an empty string) makes the icon 100% width of the enclosing container and stylable from CSS. By omitting this attribute or assigning to <code>null</code>, for example via <code>Width="@null"</code>, the default icon size will be used.

**Review by dvoituron — CHANGES_REQUESTED** [review]
Could

**Review by dvoituron — CHANGES_REQUESTED** [review]
Could you add an Unit Test to validate this new feature?

File: src/Core/Components/Icons/FluentIcon.razor.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ public partial class FluentIcon<Icon> : FluentComponentBase
 
     /// <summary />
     protected string? StyleValue => new StyleBuilder(Style)
-        .AddStyle("width", Width ?? $"{_icon.Width}px")
+        .AddStyle("width", Width ?? $"{_icon.Width}px", Width != string.Empty)
```
Comment (vnbaaij) [inline src/Core/Components/Icons/FluentIcon.razor.cs:22]: In line with @dvoituron's comment, probably better to check for IsNullOrWhitespace() ?

**Review by vnbaaij — DISMISSED** [review]

File: src/Core/Components/Icons/FluentIcon.razor.cs (hunk 2 above)
Comment (vnbaaij) [reply src/Core/Components/Icons/FluentIcon.razor.cs:22]: That is what you get when doing a coffee review on a plane 😂

**Comment by dvoituron** [comment]
Weird... Your PR doesn't seem to accept the latest `dev` changes. 
It currently contains 41 file changes when in fact there are only 4.

image

Could you check this anomaly (perhaps to close this PR and recreate a new one... I don't know)?

About the code, everything seems to be correct.

**Review by dvoituron — APPROVED** [review]

**Pull request merged**